src/include/miopen/batchnorm/problem_description.hpp

```diff
@@ -194,6 +194,20 @@ struct MIOPEN_INTERNALS_EXPORT ProblemDescription : ProblemDescriptionBase, Prob
                                               : ((in_layout == "NDHWC") && (out_layout == "NDHWC"));
     }
 
+    bool IsLayoutNCHW() const
+    {
+        if(direction == Direction::Backward)
+        {
+            return xDesc.GetLengths().size() == 4
+                       ? ((in_layout == "NCHW") && (out_layout == "NCHW") && (din_layout == "NCHW"))
+                       : ((in_layout == "NCDHW") && (out_layout == "NCDHW") &&
+                          (din_layout == "NCDHW"));
+        }
+
+        return xDesc.GetLengths().size() == 4 ? ((in_layout == "NCHW") && (out_layout == "NCHW"))
+                                              : ((in_layout == "NCDHW") && (out_layout == "NCDHW"));
+    }
+
```

Comment:
Does it support strides? Because it is possible to set `NCHW` layout with strides something like `192 1 32 4` instead of `192 48 8 1` for `NCHW` tensor `2 4 6 8`. In that case it will be actually `NHWC`.
I'm just saying that "layout" does not guarantee actual layout.
Or `NHWC` tensor of sizes `2 6 8 4` may have strides `192 8 1 48` which is actually `NCHW` and can be compatible with the algorithm.

Reply:
@CAHEK7  do you think we should also check strides to determine the layout? In conv they [just check string ](https://github.com/ROCm/MIOpen/blob/develop/src/conv/problem_description.cpp#L222)

Reply:
I thought it was safe to assume that the layout string would be set to match what the strides are representing.
I remember something similar came up with trying to figure out how to map strides to tensor layout with the graphAPI, and the [SetStrides](https://github.com/ROCm/MIOpen/blob/develop/src/tensor.cpp#L218) method would either calculate strides from a layout, or set the layout from provided strides. 

If the above is true, then I think the way it's done here, and the way it's done for conv is alright.